### ISSUE TITLE
Dynamic Workers and Queues for scaling?

### ISSUE THREAD

**Issue by DeLongShot** [koudelka/honeydew#65]
**Problem:**

I'd like to be able to scale workers & global queues as new nodes join a cluster.  For example, say I'm getting a unusually large amount of jobs, I'd like to be able to add another node and be able to start more workers on that node.

From what I can tell from the documentation and reading the code, there is no way currently to start more workers for a global queue which already has workers available.  If I have started the queue and some workers and try to start more workers on a new node, I get "error, already started".

I'm currently using global queues backed by Mnesia.  

I don't know if there is a way currently to do this.  I'd like to be able, when a new node joins the cluster, to start more workers for a queue on that node and possibly make the new node also be a "disc copy" of the queue as well.  But I'd take just being able to start more workers.

**Comment by koudelka** [comment]
Hm, I'm able to start workers on new nodes. What you're describing sounds like you might be inadvertently trying to start two sets of workers on the same node for the same queue. Are you sure you're running `Honeydew.start_workers/3` on two separate nodes?

Did you check out the ["global" example](https://github.com/koudelka/honeydew/blob/master/examples/global/global.exs)? I just tried it with two different "worker" nodes, seemed to work fine.

One thing you can't do is dynamically change the number of workers, on any one node, servicing a queue. Well, you can, but you have to stop the workers first, change the `:num` argument to `Honeydew.start_workers/3`, and start them up again. I'd love to implement real auto-scaling, but I think that's a can of worms when you get into it.

It's not documented (my bad), but the `num` argument actually indicates the number of workers that should be started _per queue node_ in the cluster. So if you specify `num: 5`, and then have two queue nodes, you'll end up with ten workers running on each of your worker nodes. When a new queue node joins the cluster, you'll get another five. If a queue node becomes uncontactable, the number reduces by five, etc..

I haven't messed around too much with the particulars of adding new nodes dynamically to the Mnesia queue, I think there's a little bit of trickiness around that, @aaronrenner did a bit of work on that recently, he might be able to fill you in.

I'm more than happy to take a look at a minimal reproduction, if this doesn't help.

**Comment by DeLongShot** [comment]
@koudelka Thank you for your answer.  You were spot-on.  I realized that I had inadvertently started workers on the same node for the same queue. 

Yeah, dynamically changing the number of workers would be great, but I understand that problem could get very complicated quickly. I figured that I could stop and start them again if I really need that.

Thanks for the heads up about the `num` argument. I may mess around with adding new queue nodes dynamically, but I think I've come up with a good compromise for now.

I'll close this issue as there's no bug associated with it and these features would be too complicated to add for now.  Thanks again for the quick answer.
